registry/auth/basic/htpasswd.go

```diff
@@ -0,0 +1,141 @@
+package basic
+
+import (
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/csv"
+	"errors"
+	"os"
+	"regexp"
+	"strings"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+// AuthenticationFailureErr - a generic error message for authentication failure to be presented to agent.
+var ErrAuthenticationFailure = errors.New("Bad username or password")
+
+// HTPasswd - holds a path to a system .htpasswd file and the machinery to parse it.
```

Comment:
Please follow [go documentation](http://blog.golang.org/godoc-documenting-go-code) guidelines. No need for the dash here.